From 9f34a9ffe5b94aee93baa5762719546e0bebc28d Mon Sep 17 00:00:00 2001
From: Kai Lueke <kailuke@microsoft.com>
Date: Fri, 15 Mar 2024 11:49:50 +0100
Subject: [PATCH] x86/boot: Reserve space for dm-verity hash storage

Reserve space in the boot sector for storing dm-verity hash data,
which is needed for Flatcar's verified boot implementation. This
adapts to the new boot sector layout in 6.16 by adding the reserved
space without interfering with the EFI stub structure.
---
 arch/x86/boot/setup.ld |  3 +++
 1 file changed, 3 insertions(+)

--- a/arch/x86/boot/setup.ld
+++ b/arch/x86/boot/setup.ld
@@ -13,6 +13,9 @@ SECTIONS
 	. = 0;
 	.bstext : {
 		*(.bstext)
+		/* Reserve space for dm-verity hash storage */
+		. = ALIGN(16);
+		dmverity_hash_space = .;
 		. = 495;
 	} =0xffffffff
